docs/providers/mongodb.md

```diff
@@ -0,0 +1,33 @@
+# MongoDB
+
+Featureform supports [MongoDB](https://www.mongodb.com/) as an Inference Store.
+
+## Implementation
+
+A MongoDB table is created for every feature. A metadata table exists within the Database as well to allow the provider to keep track of its own state. Featureform's scheduler aims to achieve consistency between MongoDBs's internal state with the user's desired state as specified in the metadata service.
```

Comment:
This paragraph is used in the other inference stores too but seems a bit wordy/convoluted. It introduces multiple elements that aren't relevant to understand the functionality. I think it is more important here to note
1. Mongodb stores values of features with their specified entity (customerID, name, timestamp) as a key
2. Featureform stores metadata on every feature registered on MongoDB, as well as its readiness for serving

Reply:
Updated to be more mongo specific